### PR TITLE
Fixed gcov not being found in certain instances.

### PR DESCRIPTION
**Pull request by nmoinvaz**
See #261

### CODE REVIEW

**Comment by nmoinvaz** [comment]
I have rebased my fork and tested again that this fixes it.

**Comment by nmoinvaz** [comment]
@thomasrockhu this one is also important as it fixes reported bug #261

File: codecov/__init__.py [hunk 1]
```diff
@@ -914,7 +914,7 @@ def main(*argv, **kwargs):
             for path in find_files(sanitize_arg("", codecov.gcov_root or root), "*.gcno", True, dont_search_here):
                 cmd = sanitize_arg("", codecov.gcov_exec or "") + " -pb " + sanitize_arg("", codecov.gcov_args or "") + " " + path
                 write("    Executing gcov (%s)" % cmd)
-                write(try_to_run(cmd))
+                write(try_to_run(cmd, shell=True))
```
Comment (thomasrockhu) [inline codecov/__init__.py:917]: @nmoinvaz my big worry here is that I think it's possible to insert malicious code here. Am I thinking about this wrong?

Comment (nmoinvaz) [reply codecov/__init__.py:917]: I think I may have solved the issue another way.. will update PR.

**Comment by nmoinvaz** [comment]
Between the time when I first created my PR #217 and when it was pulled, `try_to_run` command was converted to take a list instead of a string in https://github.com/codecov/codecov-python/commit/6fe18b3d2ec35f4201b887d3f3932ef519908d7b.

File: codecov/__init__.py (hunk 1 above)
Comment (nmoinvaz) [reply codecov/__init__.py]: Updated.

File: codecov/__init__.py [hunk 2]
```diff
@@ -925,13 +925,11 @@ def main(*argv, **kwargs):
...
-                    + " -pb "
-                    + sanitize_arg("", codecov.gcov_args or "")
-                    + " "
-                    + path
-                )
+                cmd = sanitize_arg("", codecov.gcov_exec or "").split(" ")
+                cmd.append("-pb")
+                if codecov.gcov_args:
+                    cmd.append(sanitize_arg("", codecov.gcov_args or ""))
+                cmd.append(path)
```
Comment (thomasrockhu) [inline codecov/__init__.py:932]:
```suggestion 
                cmd = [
                    *sanitize_arg("", codecov.gcov_exec or "").split(" "),
                    "-pb",
                    sanitize_arg("", codecov.gcov_args or ""),
                    path,
                  ]
```

Comment (nmoinvaz) [reply codecov/__init__.py:932]: That is how I initially had it, but my CI threw some errors with "" entries if gcov_args not supplied. It was trying to find "" as an input file.

Also I split gcov_exec due to "llvm-cov-6.0 gcov" possibly being passed in.

**Comment by nmoinvaz** [comment]
Thanks!

**Pull request merged**